source/admin_manual/migrating_mailboxes.rst

```diff
@@ -0,0 +1,314 @@
+ * :ref:`Quota <quota>` without rules (to make sure quota gets calculated, but not enforced)
+ * :ref:`Mail caching settings <mail_cache_settings>`
+ * :ref:`Attachment detection settings <setting-mail_attachment_detection_options>`
+ * :ref:`Mailbox attribute settings <setting-mail_attribute_dict>`
+ * :ref:`Compression <zlib_plugin>` and :ref:`encryption <mail_crypt_plugin>` settings
+ * :ref:`NFS related settings <nfs>`
+
+Sometimes, if your production configuration file has complex authentication flows, push notifications or other settings that might cause unwanted effects during migration,
+you can make a separate migration configuration file. To use this configuration file, you can store it for example as ``/etc/dovecot/dovecot-migration.conf``,
+and use it with ``doveadm -c /etc/dovecot/dovecot-migration.conf sync``.
+
+Other settings to use::
+
+  # this is sometimes needed with some broken servers
+  #dsync_features = empty-header-workaround
+
+  # Read multiple mails in parallel, improves performance
+  mail_prefetch_count = 20
+
+  # If you have very large mailboxes, you might want to tune this
+  #dsync_commit_msgs_interval = 100 # default in v2.2.30+
+
+  # If you need to hash other headers for whatever reason, you can change this
+  #dsync_hashed_headers = Date Message-ID
+
+  # Avoid useless IO
+  mail_fsync = never
+
+Testing
+=======
+
+It is a very good idea to do some test migrations first using test accounts to ensure everything works.
+
+When migrating mails from another server, you should make sure that these are preserved:
+
+ #. Message flags
+
+    * Lost flags can be really annoying, you most likely want to avoid it.
+
+ #. Message UIDs and UIDVALIDITY value
+
+    *  If UIDs are lost, at the minimum clients' message cache gets cleaned and messages are re-downloaded as new.
+    *  Some IMAP clients store metadata by assigning it to specific UID, if UIDs are changed these will be lost.
+
+ #. Mailbox subscription list
+
+.. _migrating_mailboxes_dovecot:
+
+Migrating mailboxes from another Dovecot
+========================================
+
+Preparations
+------------
+
+If the old system is running v2.1.14+ , you can use doveadm protocol to migrate your mails. If the old system is older, see :ref:`migrating_mailboxes_imapc`.
+
+First, setup doveadm service on old server::
+
+  service doveadm {
+     inet_listener {
+       port = 12354
+     }
+  }
+
+  doveadm_password = supersecret
+
+Then put doveadm password to new server::
+
+  doveadm_password = supersecret
+
+Now you are ready to migrate.
+
+Executing migration
+-------------------
+
+If you want to avoid any changes to your source system, unidirectional synchronization is recommended.
+
+To migrate users, use::
+
+  doveadm backup -Ru username tcp:host:port
+
+If you are experiencing problems, run::
+
+  doveadm -D backup -Ru username tcp:host:port
+
+This will enable debug logging.
+
+You can run the command again to perform incremental updates.
```

Comment:
Add something like: The `doveadm backup` command forces the destination to look exactly like the source, deleting mails and mailboxes if necessary. If it's possible that the destination already has new mails (or other changes), use `doveadm sync -1` instead::

   doveadm sync -1Ru username tcp:host:port